eng/common/scripts/FilterPoliCheckResults.ps1

```diff
@@ -6,35 +6,42 @@ This script will read data speciefied in one or more PoliCheckAllowList.yml file
 It then reamoves all allwed entries from the PoliCheckResult 
 .PARAMETER PoliCheckResultFilePath
 The Path to the PoliCheck Result. Usually named PoliCheck.sarif
-.PARAMETER ServiceDirtectory
+.PARAMETER ServiceDirectory
 If the PoliCheck scan is scoped to a particular service provide the ServiceDirectory
+.PARAMETER AllowListLocation
+A path to a folder containing yml defined policheck allowlist
 .EXAMPLE
-PS> ./FilterPoliCheckResults.ps1 -PoliCheckResultFilePath .\PoliCheck.sarif
+PS> ./FilterPoliCheckResults.ps1 -PoliCheckResultFilePath .\PoliCheck.sarif -ServiceDirectory <servicedirname> -AllowListLocation <location>
 #>
 [CmdletBinding()]
 param(
   [Parameter(Mandatory=$true)]
   [String] $PoliCheckResultFilePath,
-  [String] $ServiceDirtectory
+  [Parameter(Mandatory=$true)]
+  [String] $ServiceDirectory,
+  [Parameter(Mandatory=$true)]
+  [String] $AllowListLocation
 )
 
 . "${PSScriptRoot}\logging.ps1"
 
-$RepoRoot = Resolve-Path -Path "${PSScriptRoot}\..\..\..\"
-$PathToAllowListFiles = Join-Path $RepoRoot $ServiceDirtectory
-$PolicCheckAllowListFiles = Get-ChildItem -Path $PathToAllowListFiles -Recurse -File -Include "PoliCheckAllowList.yml"
+# Install Powershell Yaml
+$ProgressPreference = "SilentlyContinue"
+$toolsFeed = "https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-tools/nuget/v2"
+Register-PSRepository -Name azure-sdk-tools-feed -SourceLocation $toolsFeed -PublishLocation $toolsFeed -InstallationPolicy Trusted -ErrorAction SilentlyContinue
+Install-Module -Repository azure-sdk-tools-feed powershell-yaml
+
+$allowListFilePath = Join-Path $AllowListLocation "${ServiceDirectory}.yml"
```

Comment:
For my understanding: where is this file coming from? Do we need to worry about cross-language naming overlaps and/or service directory [paths like](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/eventhub/event-hubs) `eventhub/event-hubs` causing issues?

Reply:
That won't be a problem there is a directory for each language.

Reply:
What about ServiceDirectory values that aren't valid filenames? I guess I was wrong in that there are no examples of these in the tier 1 language repos, though there are many in the go repo. I think we could either split it in `/`, or just wait and see if we ever add policheck to go and deal with it then.

Reply:
Yes, will probably spit it when the time comes to add go.